private/buf/cmd/buf/command/push/push.go

```diff
@@ -199,6 +220,33 @@ func push(
 	if err != nil {
 		return nil, err
 	}
+	modulePin, err := push(ctx, container, clientConfig, moduleIdentity, builtModule, flags)
+	if err != nil {
+		// We rely on Push* returning a NotFound error to denote the repository is not created.
+		// This technically could be a NotFound error for some other entity than the repository
+		// in question, however if it is, then this Create call will just fail as the repository
+		// is already created, and there is no side effect. The 99% case is that a NotFound
+		// error is because the repository does not exist, and we want to avoid having to do
+		// a GetRepository RPC call for every call to push --create.
```

Comment:
Why not instead do an invocation to `CreateRepository` RPC always if the `--create` flag is passed.

That should return either an OK (repo created), or an AlreadyExists error (I assume this `--create` flag is idempotent?), so with any of those two results, we continue with the push, otherwise, we fail the push command.

Reply:
Actually just thought about it again (after implementing it and almost pushing it) - its better to first do the `push` call, because in the 99% case, this will result in only one RPC, i.e. the only time that `push` should fail is on the first instance where the repository is not created. For the remainder of the lifetime of the module, `push --create` will be unnecessarily be doing two RPCs instead of one, with the first RPC failing.

Reply:
Alright, I guess this is a tradeoff to protect network usage against users adding this flag always to CIs, when in reality they need to run it just once.